test/python/transpiler/test_calibrationbuilder.py

```diff
@@ -233,76 +177,269 @@ def test_non_native_cr_with_target(self):
+    #     else:
+    #         qubits = tuple(reversed(test_qubits))
+
+    #     cr_schedule = target["ecr"][qubits].calibration
+
+    #     qc = circuit.QuantumCircuit(2)
+    #     qc.rzx(theta, qubits[0], qubits[1])
+    #     _pass = RZXCalibrationBuilder(target=target)
+    #     test_qc = PassManager(_pass).run(qc)
+
+    #     ref_sched = self.build_stretched_ecr_pulse_forward(
+    #         backend=backend,
+    #         theta=theta,
+    #         u0p_play=self.get_cr_play(cr_schedule, "CR90p_u"),
+    #         d1p_play=self.get_cr_play(cr_schedule, "CR90m_u"),
+    #         u0m_play=self.get_cr_play(cr_schedule, "CR90p_d"),
+    #         d1m_play=self.get_cr_play(cr_schedule, "CR90m_d"),
+    #     )
+
+    #     self.assertEqual(schedule(test_qc, backend), target_qobj_transform(ref_sched))
+
+
+@ddt
+class TestRZXCalibrationBuilder(TestCalibrationBuilder):
+    """Test RZXCalibrationBuilder."""
 
-        _pass = RZXCalibrationBuilder(self.inst_map)
+    @data(-np.pi / 4, 0.1, np.pi / 4, np.pi / 2, np.pi)
+    def test_rzx_calibration_cr_pulse_stretch(self, theta: float):
+        """Test that cross resonance pulse durations are computed correctly."""
+        # TODO - need to do this with BackendV2 and FakeSherbrooke
+        backend = FakeHanoi()
+        inst_map = backend.defaults().instruction_schedule_map
+        cr_schedule = inst_map.get("cx", (0, 1))
+        with builder.build() as test_sched:
+            RZXCalibrationBuilder.rescale_cr_inst(self.u0p_play(cr_schedule), theta)
+
+        self.assertEqual(
+            test_sched.duration, self.compute_stretch_duration(self.u0p_play(cr_schedule), theta)
+        )
+
+    @data(-np.pi / 4, 0.1, np.pi / 4, np.pi / 2, np.pi)
+    def test_rzx_calibration_rotary_pulse_stretch(self, theta: float):
+        """Test that rotary pulse durations are computed correctly."""
+        # TODO - need to do this with BackendV2 and FakeSherbrooke
+        backend = FakeHanoi()
+        inst_map = backend.defaults().instruction_schedule_map
+        cr_schedule = inst_map.get("cx", (0, 1))
+        with builder.build() as test_sched:
+            RZXCalibrationBuilder.rescale_cr_inst(self.d1p_play(cr_schedule), theta)
+
+        self.assertEqual(
+            test_sched.duration, self.compute_stretch_duration(self.d1p_play(cr_schedule), theta)
+        )
+
+    def test_ecr_cx_forward(self):
+        """Test that correct pulse sequence is generated for native CR pair."""
+        # Sufficiently large angle to avoid minimum duration, i.e. amplitude rescaling
+        theta = np.pi / 4
+
+        qc = circuit.QuantumCircuit(2)
+        qc.rzx(theta, 0, 1)
+
+        backend = FakeHanoi()
+        inst_map = backend.defaults().instruction_schedule_map
+        _pass = RZXCalibrationBuilder(inst_map)
         test_qc = PassManager(_pass).run(qc)
 
-        duration = self.compute_stretch_duration(self.u0p_play, theta)
+        cr_schedule = inst_map.get("cx", (0, 1))
```

Comment:
You can just call `build_stretched_ecr_pulse_forward` method from the superclass to generate reference schedule.

Reply:
Replaced builder code with superclass methods.